cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -626,4 +626,37 @@ private Path findFirstRecursive(Path dir, Predicate<Path> filter, boolean recurs
     return null;
   }
 
+  @Override
+  public List<Path> getChildrenInDir(Path dir, Predicate<Path> filter) {
+    List<Path> files = new ArrayList<>();
+    try (Stream<Path> childStream = Files.list(dir)) {
+      Iterator<Path> iterator = childStream.iterator();
+      while (iterator.hasNext()) {
+        Path child = iterator.next();
+        if (filter.test(child)) {
+          files.add(child);
+        }
+      }
+    } catch (IOException e) {
+      throw new IllegalStateException("Failed to iterate through " + dir, e);
+    }
+    return files;
+  }
+
+  @Override
+  public Path findExistingFile(String fileName, List<Path> searchDirs) {
+
+    for (Path dir : searchDirs) {
+      Path filePath = dir.resolve(fileName);
+      try {
+        if (Files.exists(filePath)) {
+          return filePath;
+        }
+      } catch (SecurityException e) {
+        throw new IllegalStateException("SecurityException while checking file existence.");
```

Comment:
https://github.com/devonfw/IDEasy/blob/main/documentation/coding-conventions.asciidoc#catching-and-handling-exceptions